### PR TITLE
lib/log: //echo/printf

### PR DESCRIPTION
**Pull request by gaelicWizard**
## Description
- Replace `echo -e` with `printf` in `_bash-it-log-message()`.
- `local` positional parameters to allow for defaults.
- Use new `composure.sh` feature to reduce load order dependency.
- Use `if`/`then` properly.
- Clean up use of `$BASH_IT_LOG_PREFIX` slightly (eliminate duplicate colons).

## Motivation and Context
Misuse of `&&`/`||` can cause unexpected return status, so I'm just fixing to use `if`/`then` wherever I see them. `echo -e` is unreliable and has differing behaviour in different contexts, such as in subshells and when `xpg_echo` is enabled. `printf` does exactly what it says on the tin. The parameter `$message` was leaking from every log function, so I `local`'d it. Alsö, `local` the rest of the parameters and fill in a default color, default log level, and strip duplicate colons from the message and level. 

## How Has This Been Tested?
Now that #1902 is merged, I can start a whole shell straight off `master`!! So, anyway, I have my `$BASH_IT_LOG_LEVEL` set to max in my `~/.bashrc` so I see all the messages and everything still works. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [x] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

File: lib/log.bash [hunk 1]
```diff
@@ -45,7 +45,7 @@ function _bash-it-log-prefix-by-path() {
 
 function _has_colors() {
 	# Check that stdout is a terminal, and that it has at least 8 colors.
-	[[ -t 1 && "${_bash_it_available_colors:=$(tput colors 2> /dev/null)}" -ge 8 ]]
+	[[ -t 1 && "${CLICOLOR:=$(tput colors 2> /dev/null)}" -ge 8 ]]
```
Comment (NoahGorny) [inline lib/log.bash:48]: why this change?

Comment (gaelicWizard) [reply lib/log.bash]: That snuck in from another branch! Fixed.

**Review by NoahGorny — APPROVED** [review]
LGTM
